### PR TITLE
fix(database): add prefix to avoid join column name conflicts

### LINKED ISSUE #18085: Adding properties to entities named "<entity name>Id" cause issues when nesting relationship fields
### Required System information

- Node.js version: 16.6.1
- NPM version: 8.2.0
- Strapi version: 4.9.2
- Database: Mysql 2.18.1
- Operating system: mac/windows/linux
- Is your project Javascript or Typescript: Javascript

### Describe the bug
Adding a property to an entity with the name of "<entity name>Id" that also has a relationship property to the same entity, when getting results from the api if you try to read the value of <entity name>Id of the parent from inside the child the value will be the row ID of the child.

### Steps to reproduce the behavior

- Create an entity named 'shirt'
- Add a text prop named 'shirtId'
- Save
- Add a one to many relational field to shirt named variants for the many and variantOf for the one.
- Create a shirt with a shirtId = 'abc'
- Create a shirt with a shirtId = 'xyz'
- Add the second shirt as a variant of the first shirt

 When pulling results from the api:
- The parent shirt's shirtId = "abc" and ID of 1 
- The parent has a variant shirt with a shirtId of "xyz" and ID of 2
- The variant shirt has a variantOf property to reference it's parent 
- The variant shirt's variantOf shirt will have a shirtId of "2" (the ID of the variant) and an ID of 1

### Expected behavior
The <entity name>Id field should be correct when looking at the parent data from the child

### Additional context

If this shirtId property is changed to something like "shirtUid" the bug does not happen.

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Uses an alias for the join column name to prevent it conflicting with a column name on the main table.

### Why is it needed?

If the join column name was the same as a column name in the main content type's table, it was always overwritten.

So for example, in #18085 , if we create a `shirt` content type with a `shirt_id` attribute, and then have a relation to it, the join was using `shirt_id` as the join column name, overwriting the `shirt_id` attribute of `shirt`.

Technically this is still the case, but it's extremely unlikely someone would have an attribute in a content type named `__strapi{ct name}_id` unless they were intentionally trying to encounter this issue. Regardless, In Strapi 5 it will be an officially reserved prefix.

This should be safe for long identifiers in the db, because any existing database with a join table will include an index named
`{ct name}_{attribute name}_unique`
Which means, given a minimum 1 char for ct name, we have len(`1__unique`) = 9 > len(`__strapi`) and should not hit any limit that isn't already being hit.

### How to test it?

see #18085

### Related issue(s)/PR(s)

Fixes #18085

DX-1178

### CODE REVIEW

File: packages/core/database/src/query/helpers/populate/apply.ts [hunk 1]
```diff
@@ -87,6 +87,11 @@ const XtoOne = async (
 
     const alias = qb.getAlias();
     const joinColAlias = `${alias}.${joinColumnName}`;
+    // We must select the join column id, however whatever it is named will overwrite an attribute of the same name
+    // Therefore, we will use something unlikely to conflict (strapi_{content type name}_id)
+    // TODO: ...and completely restrict the strapi_ prefix for an attribute name in the future
+    const joinColRenameAs = `strapi_${joinColumnName}`;
```
Comment (alexandrebodin) [inline packages/core/database/src/query/helpers/populate/apply.ts:93]: Looks like this should not be done only for XtoOne ?

**Comment by innerdvations** [comment]
@pwizla This is targeting v4, and could technically be considered a breaking change although it is very unlikely to affect anyone, but we should probably document somewhere that you can't use `__strapi_{content type name}_id` as the name of an attribute just so it's not lost and forgotten.

**Comment by innerdvations** [comment]
This somehow breaks pagination in review workflows, investigating it now.

**Review by christiancp100 — APPROVED** [review]
QAed this and works as expected!!

**Pull request merged**